packages/web-app-app-store/tests/unit/LayoutContainer.spec.ts

```diff
@@ -0,0 +1,52 @@
+import LayoutContainer from '../../src/LayoutContainer.vue'
+import { defaultComponentMocks, defaultPlugins, mount, nextTicks } from 'web-test-helpers'
+import { useAppsStore } from '../../src/piniaStores'
+
+const selectors = {
+  loadingSpinner: '#app-loading-spinner',
+  routerView: '[data-testid="app-store-router-view"]'
+}
+
+describe('LayoutContainer', () => {
+  it('shows a loading spinner while apps are loading from repositories', () => {
+    const { wrapper } = getWrapper({ keepLoadingApps: true })
+    expect(wrapper.find(selectors.loadingSpinner).exists()).toBeTruthy()
+    expect(wrapper.find(selectors.routerView).exists()).toBeFalsy()
+  })
+  it('renders the router view when loading apps is done', async () => {
+    const { wrapper } = getWrapper({})
+    await nextTicks(2)
```

Comment:
Hmm this feels a bit brittle, I guess you wait for the 500 ms to pass here? Maybe you can return the promise in the wrapper and await it here?

Reply:
I don't want to let the 500ms pass. I want to stay in that timeframe so that a loading state is still active. Maybe still better to not do that with a timeout.

Reply:
Ah lol, sorry for my confusion. I thought it was the other test. The other test passes a `keepLoadingApps: true` to the wrapper. Only that one has the 500ms timeout. This one here doesn't have a loading delay.